src/features/todos.ts

```diff
@@ -1,9 +1,26 @@
 import { Todo } from '../types/Todo';
 
-export const actions = {};
+type SetTodosAction = {
+  type: 'todos/SET_TODOS',
+  payload: Todo[]
+};
+
+type Action = SetTodosAction;
+
+export const setTodos = (todos: Todo[]): SetTodosAction => ({
+  type: 'todos/SET_TODOS',
+  payload: todos,
+});
+
+export const actions = { setTodos };
 
-const todosReducer = (): Todo[] => {
-  return [];
+const todosReducer = (todos = [], action: Action): Todo[] => {
+  switch (action.type) {
+    case 'todos/SET_TODOS':
+      return [...action.payload];
+    default:
+      return [...todos];
+  }
```

Comment:
do not rewrite state in dafault case - it way cause troubles
```suggestion
const todosReducer = (state= [], action: Action): Todo[] => {
  switch (action.type) {
    case 'todos/SET_TODOS':
      return [...action.payload];
    default:
      return state;
  }
```